### PR TITLE
Get Establishment URNs by region

### PR DESCRIPTION
**Pull request by dneed-nimble**

### CODE REVIEW

File: TramsDataApi.Test/UseCases/GetEstablishmentURNsByRegionTests.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            _establishmentGateway = new Mock<IEstablishmentGateway>();
+            _useCase = new GetEstablishmentsByRegion(_establishmentGateway.Object);
+        }
+
+        [Fact]
+        public void GetEstablishmentURNsByRegion_ReturnsNull_WhenNoEstablishmentsAreFound()
+        {
+            _establishmentGateway.Setup(gateway => gateway.GetURNsByRegion(Regions)).Returns(() => null);
+
+            _useCase.Execute(Regions).Should().BeNull();
```
Comment (jordan-beard) [inline TramsDataApi.Test/UseCases/GetEstablishmentURNsByRegionTests.cs:47]: Small thing: as discussed yesterday perhaps it'd be better to return an empty list.

Comment (dneed-nimble) [reply TramsDataApi.Test/UseCases/GetEstablishmentURNsByRegionTests.cs:47]: Ah yeah, thanks!

File: TramsDataApi/Gateways/EstablishmentGateway.cs [hunk 2]
```diff
@@ -17,6 +18,14 @@ public Establishment GetByUkprn(string ukprn)
         { 
             return _dbContext.Establishment.FirstOrDefault(e => e.Ukprn == ukprn);
         }
+        public IEnumerable<int> GetURNsByRegion(IEnumerable<string> regions)
```
Comment (jordan-beard) [inline TramsDataApi/Gateways/EstablishmentGateway.cs:21]: It looks like you're following convention here but I'm surprised to see these aren't async.

File: TramsDataApi.Test/Integration/EstablishmentsIntegrationTests.cs [hunk 3]
```diff
@@ -225,11 +240,12 @@ public async Task CanSearchEstablishmentsByPartialUkprn()
             result.Should().BeEquivalentTo(expected);
         }
 
-        private EstablishmentResponse AddTestData(int urn)
+        private EstablishmentResponse AddTestData(int urn, string? region)
```
Comment (jordan-beard) [inline TramsDataApi.Test/Integration/EstablishmentsIntegrationTests.cs:243]: Small thing: could make this an optional parameter.

File: TramsDataApi/Controllers/EstablishmentsController.cs [hunk 4]
```diff
@@ -47,6 +49,22 @@ public ActionResult<EstablishmentResponse> GetByUkprn(string ukprn)
             _logger.LogDebug(JsonSerializer.Serialize<EstablishmentResponse>(establishment));
             return Ok(establishment);
         }
+        [HttpGet]
+        [Route("establishment/regions")]
+        public ActionResult<IEnumerable<int>> GetURNsByRegion([FromQuery] string[] regions)
```
Comment (jordan-beard) [inline TramsDataApi/Controllers/EstablishmentsController.cs:54]: Do we have any controller tests? Might be worth adding a couple for this.

File: TramsDataApi/Gateways/EstablishmentGateway.cs (hunk 2 above)
Comment (dneed-nimble) [reply TramsDataApi/Gateways/EstablishmentGateway.cs:21]: I agree it is strange. Happy to make the change if you think it's worth it?

File: TramsDataApi.Test/Integration/EstablishmentsIntegrationTests.cs (hunk 3 above)
Comment (dneed-nimble) [reply TramsDataApi.Test/Integration/EstablishmentsIntegrationTests.cs]: Agree, have changed.

File: TramsDataApi/Controllers/EstablishmentsController.cs (hunk 4 above)
Comment (dneed-nimble) [reply TramsDataApi/Controllers/EstablishmentsController.cs:54]: Added some controller tests

**Review by jordan-beard — APPROVED** [review]

**Pull request merged**